src/pybind/mgr/cephadm/serve.py

```diff
@@ -112,6 +114,39 @@ def serve(self) -> None:
             self.log.debug("serve loop wake")
         self.log.debug("serve exit")
 
+    def _check_certificates(self) -> None:
+        for d in self.mgr.cache.get_daemons_by_type('grafana'):
+            cert = self.mgr.get_store(f'{d.hostname}/grafana_crt')
+            key = self.mgr.get_store(f'{d.hostname}/grafana_key')
+            if (not cert or not cert.strip()) and (not key or not key.strip()):
```

Comment:
if an empty string is passed, nothing to check: this would probably answer my previous question, and iiuc it applies also for use cases where the two keys are not defined at all, am I right?

- case 1: no grafana_{crt,key} is defined -> the config will be `http` by default
- case 2: I set those keys, we have code (L126 - L128) that ensure the provided info are not `malformed`.

Reply:
> if an empty string is passed, nothing to check: this would probably answer my previous question, and iiuc it applies also for use cases where the two keys are not defined at all, am I right?
> 
> * case 1: no grafana_{crt,key} is defined -> the config will be `http` by default
> * case 2: I set those keys, we have code (L126 - L128) that ensure the provided info are not `malformed`.

If no `grafana_{crt,key}` is defined (empty values) cephadm will generate crt/key automatically and use them to deploy grafana. So as per the current (and legacy) code we will be always using certificates for grafana (so no plain http is used I guess).

Reply:
ack, I'll check this part in one of the OSP CI jobs, it's a good practice and I understand the purpose of generating certificates. 
I'll let you know once I have the chance to check one of the OSP jobs, but in general I'm ok with this change.